cmd/internal/shell/core.go

```diff
@@ -49,13 +49,23 @@ func Get() Shell {
 //   `%F{cyan}wash ${prompt_path}%F{green} ❯%f `
 // with substitution for shell-specific portions of the function.
 func preparePrompt(cyan, green, reset, assign string) string {
+	// If PWD is a subdirectory of $W, then strip $W/. By removing the slash if it matches, we should
+	// prevent any further matches with absolute paths.
+	// If prompt path equals $W, set to '.'.
+	// If prompt path is a subdirectory of $HOME, then replace with '~'.
 	return `
 function prompter() {
 	local prompt_path
-	if [ -x "$(command -v realpath)" ]; then
-		prompt_path=$(realpath --relative-base=$W "$(pwd)")
-	else
-		prompt_path=$(basename "$(pwd)")
+
+	if [ -v W ]; then
+		prompt_path=${PWD/#$W\//}
+		if [ "${prompt_path}" = "${W}" ]; then
+			prompt_path=.
+		fi
+	fi
+
+	if [ -v HOME ]; then
+		prompt_path=${prompt_path/#$HOME/\~}
```

Comment:
When will it ever be a subdirectory of `HOME` and `W`? If never, then this would be clearer if it were an `elseif`

Reply:
Yeah, I may be trying to make it shorter at the cost of readability.

Reply:
It's a little challenging to make the right logic. That's why I put the comment at the top. I'm relying on string substitution to strip the leading `/` so later substitution won't match.

Reply:
I think it might be simpler if the comment were written in terms of prompt_path (instead of PWD _and_ prompt_path). Something like

```
// Here's how to calculate prompt_path.
//
//. prompt_path = PWD
//  If prompt_path is subdir of W, then
//     prompt_path == <part_after_W>
//.    if prompt_path == <empty> then prompt_path = "."
//. else if prompt_path is subdir of HOME, then
//.   prompt_path = ~/<part_after_HOME>
//
// return prompt_path
```

Reply:
Ok, I can update that.

Reply:
Shouldn't the implementation reflect that?

Reply:
I'll work on a new description. What you wrote doesn't precisely match what's happening.

Reply:
Ok.